src/chrome/locale/en/https-everywhere.dtd

```diff
@@ -25,6 +25,9 @@
 <!ENTITY https-everywhere.prefs.reset_defaults "Reset to Defaults">
 <!ENTITY https-everywhere.prefs.reset_defaults_message "This will reset each ruleset to its default state. Continue?">
 
+<!ENTITY https-everywhere.cancel.he_blocking "HTTPS Everywhere is currently blocking all unencrypted requests.">
+<!ENTITY https-everywhere.cancel.he_blocking_explainer "This page could not be served over HTTPS, either because no HTTPS page is available, or because the HTTPS version of this website is being blocked. If you wish to view the unencrypted version of this page, you can still do so by disabling the 'Block all unencrypted requests' option in your HTTPS Everywhere extension. Be aware that disabling this option could make your browser vulnerable to network-based downgrade attacks on websites you visit.">
```

Comment:
I think it would be helpful to also inform the user that they followed a link or entered a URL that was HTTP, and HTTPS Everywhere tried to rewrite it and failed. Something like:

"HTTPS noticed you were navigating to a non-HTTPS page, and tried to send you to the HTTPS version instead. The HTTPS version is unavailable. Most likely this site does not support HTTPS, but it is also possible that an attacker is blocking the HTTPS version. If you wish to view the unencrypted version of this page, you can still do so by disabling the 'Block all unencrypted requests' option in your HTTPS Everywhere extension."

I would leave out the "vulnerable to network-based downgrade attacks" since it's not strictly accurate. If you're clicking a non-HTTPS link, and HTTPS Everywhere tries a speculative upgrade, failing that upgrade is not a downgrade attack; it's just the status quo.

Reply:
> If you're clicking a non-HTTPS link, and HTTPS Everywhere tries a speculative upgrade, failing that upgrade is not a downgrade attack; it's just the status quo.

Explicit network downgrade attacks ala `sslstrip` will work when `Block all unencrypted requests` is disabled, but will not work when it's enabled.  Saying that disabling this option will make you vulnerable to these type of attacks is strictly accurate, or am I missing something?

I like the amended text otherwise.

Reply:
Let's talk specifically about `sslstrip` attacks then (and we can handily link to sslstrip). I don't really consider sslstrip a downgrade attack: it just prevents you from receiving an (insecure) redirect.